eng/pipelines/apiview-review-gen-swagger.yml

```diff
@@ -27,7 +27,7 @@ jobs:
   - script: >
       dotnet tool install
       Azure.Sdk.Tools.SwaggerApiParser
-      --version 1.0.7-dev.20230227.3
+      --version $(SWAGGER_API_PARSER_VERSION)
```

Comment:
As we chatted while I'm ok with this for now we should have a larger conversation at some point about the right place to pull this version from. I would like to have it be shared with all the places that consume it and not duplicated in variables across different pipelines.